src/clj/game/cards.clj

```diff
@@ -27,6 +27,43 @@
   triggers"
   {:effect (req (toast state :corp "Reminder: You have unrezzed cards with \"when turn begins\" abilities." "info"))})
 
+(declare reorder-final) ;forward reference since reorder-choice and reorder-final are mutually recursive
+(defn reorder-choice
+  "Generates a recursive prompt structure for cards that do reordering (Indexing, Making an Entrance, etc.)
+
+  reorder_side is the side to be reordered, i.e. :corp for Indexing and Precognition.
+  wait_side is the side that has a wait prompt while ordering is in progress, i.e. :corp for Indexing and Spy Camera.
+
+  This is part 1 - the player keeps choosing cards until there are no more available choices. A wait prompt should
+  exist before calling this function. See Indexing and Making an Entrance for examples on how to call this function."
+
+  [reorder_side wait_side remaining chosen n original]
+  {:prompt (str "Choose a card to move next onto " (if (= reorder_side :corp) "R&D" "your Stack"))
+   :choices remaining
+   :delayed-completion true
+   :effect (req (let [chosen (cons target chosen)]
+                  (if (< (count chosen) n)
+                    (continue-ability state side (reorder-choice reorder_side wait_side (remove-once #(not= target %) remaining)
+                                                                 chosen n original) card nil)
+                    (continue-ability state side (reorder-final reorder_side wait_side chosen original) card nil))))})
+
+(defn- reorder-final
+  "Generates a recursive prompt structure for cards that do reordering (Indexing, Making an Entrance, etc.)
+  This is part 2 - the player is asked for final confirmation of the reorder and is provided an opportunity to start over."
+
+  [reorder_side wait_side chosen original]
+  {:prompt (str "The top cards of " (if (= reorder_side :corp) "R&D" "your Stack")
+                " will be " (clojure.string/join  ", " (map :title chosen)) ".")
+   :choices ["Done" "Start over"]
+   :delayed-completion true
+   :effect (req (if (= target "Done")
+                  (do (swap! state update-in [(if (= reorder_side :corp) :corp :runner) :deck]
```

Comment:
Instead of `(if (= reorder_side :corp) :corp :runner)`, can you do `reorder_side`? You do this in a few places.